### PR TITLE
Initial API commit

### PR DESCRIPTION
**Pull request by smsliman**

### CODE REVIEW

File: API/main.py [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+import io
+from io import BytesIO
+import sys
+import base64
+import math
+import cv2
+import numpy as np
+from PIL import Image
+
+# from PIL.Image import core as _imaging
```
Comment (nkalupahana) [inline API/main.py]: no

File: API/main.py [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+                            wtexts.append(block["Text"])
+                        else:
+                            if (
+                                block["Geometry"]["BoundingBox"]["Width"]
+                                <= block["Geometry"]["BoundingBox"]["Height"]
+                                and block["Text"] not in htexts
+                            ):
+                                htexts.append(block["Text"])
+    print("Blocks detected: " + str(blocks))
+    # singles = "NONE" if not stexts else str(stexts)
```
Comment (nkalupahana) [inline API/main.py]: no

File: API/main.py [hunk 3]
```diff
@@ -0,0 +1,386 @@
...
+    left = width * box["Left"]
+    top = height * box["Top"]
+    draw.rectangle(
+        [left, top, left + (width * box["Width"]), top + (height * box["Height"])],
+        fill=boxColor,
+    )
+
+
+# Displays information about a block returned by text detection and text analysis
+def DisplayBlockInformation(block):
```
Comment (nkalupahana) [inline API/main.py]: no

File: API/main.py [hunk 4]
```diff
@@ -0,0 +1,386 @@
...
+    stexts = [text.replace(" ", "") for text in stexts]
+    wtexts = [text.replace(" ", "") for text in wtexts]
+    htexts = [text.replace(" ", "") for text in htexts]
+    return {
+        "statusCode": 200,
+        "body": json.dumps({"single": stexts, "tall": htexts, "wide": wtexts}),
+    }
+
+
+def ShowBoundingBox(draw, box, width, height, boxColor):
```
Comment (nkalupahana) [inline API/main.py]: no

File: API/main.py [hunk 5]
```diff
@@ -0,0 +1,386 @@
...
+
+    left = width * box["Left"]
+    top = height * box["Top"]
+    draw.rectangle(
+        [left, top, left + (width * box["Width"]), top + (height * box["Height"])],
+        outline=boxColor,
+    )
+
+
+def ShowSelectedElement(draw, box, width, height, boxColor):
```
Comment (nkalupahana) [inline API/main.py]: no

File: API/main.py [hunk 6]
```diff
@@ -0,0 +1,386 @@
...
+            print("Not selected")
+
+    if "Page" in block:
+        print("Page: " + block["Page"])
+    print()
+
+
+def process_request(body):
+
+    # Get the document from S3
```
Comment (nkalupahana) [inline API/main.py]: no

File: API/main.py [hunk 7]
```diff
@@ -0,0 +1,386 @@
...
+    # image=Image.open(stream)
+    # Analyze the document
+    client = boto3.client('textract')
+
+    # image_binary = stream.getvalue()
+    response = client.analyze_document(
+        Document={"Bytes": stream}, FeatureTypes=["TABLES", "FORMS"]
+    )
+
+    # Alternatively, process using S3 object
```
Comment (nkalupahana) [inline API/main.py]: no

File: API/main.py [hunk 8]
```diff
@@ -0,0 +1,386 @@
...
+    # response = client.analyze_document(
+    #    Document={'S3Object': {'Bucket': bucket, 'Name': document}},
+    #    FeatureTypes=["TABLES", "FORMS"])
+
+    # Get the text blocks
+    blocks = response["Blocks"]
+    # width, height = org_image.size
```
Comment (nkalupahana) [inline API/main.py]: no

File: API/main.py [hunk 9]
```diff
@@ -0,0 +1,386 @@
...
+        # for polygon in block['Geometry']['Polygon']:
+        #    points.append((width * polygon['X'], height * polygon['Y']))
+        # draw.polygon((points), outline='blue')
+
+    # Display the image
+    # image.show()
+    return blocks
+
+
+# def process_text_analysis(bucket, document):
```
Comment (nkalupahana) [inline API/main.py]: no no

File: API/main.py [hunk 10]
```diff
@@ -0,0 +1,386 @@
...
+#     #Get the text blocks
+#     blocks=response['Blocks']
+#     width, height =image.size
+#     draw = ImageDraw.Draw(image)
+#     print ('Detected Document Text')
+
+#     # Create image showing bounding box/polygon the detected lines/text
+#     for block in blocks:
+
+#         DisplayBlockInformation(block)
```
Comment (nkalupahana) [inline API/main.py]: no no no

File: API/main.py [hunk 11]
```diff
@@ -0,0 +1,386 @@
...
+
+# def main():
+
+#     bucket = 'wordbites-data'
+#     document = 'test.jpg'
+#     block_count=process_text_analysis(bucket,document)
+#     print("Blocks detected: " + str(block_count))
+
+
+# if __name__ == "__main__":
```
Comment (nkalupahana) [inline API/main.py]: no no no no

File: API/main.py (hunk 1 above)
Comment (nkalupahana) [reply API/main.py]: also spacing

**Review by nkalupahana — CHANGES_REQUESTED** [review]

**Comment by zspherez** [comment]
@verizon this u?

File: API/main.py (hunk 11 above)
Comment (zspherez) [reply API/main.py]: yes yes yes yes

**Review by nkalupahana — APPROVED** [review]

**Comment by zspherez** [comment]
@bluejeans this u?

**Pull request merged**